src/compiler/checker.ts

```diff
@@ -12953,6 +12984,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     links.instantiations = new Map<string, Type>();
                     links.instantiations.set(getTypeListId(typeParameters), type);
                 }
+                if (type === intrinsicMarkerType && symbol.escapedName === "BuiltinIteratorReturn") {
+                    type = strictBuiltinIteratorReturn ? undefinedType : anyType;
+                }

```

Comment:
Can this be done in the same way as `IntrinsicTypeKind` like `IntrinsicTypeKind.NoInfer`, rather than hardcoding the name in a couple of places?

Reply:
No. `IntrinsicTypeKind.NoInfer` (and all of the other `IntrinsicTypeKind` entries) indicate a generic type mapping, e.g. `type NoInfer<T> = intrinsic` maps `T` to something else, which in that instance is a substitution type. Since `BuiltinIteratorReturn` is not generic it isn't picked up by `getTypeAliasInstantiation`, so never hits those code paths.

Reply:
If anything, it would exist in its own map of non-generic `intrinsic` types, but since it would be the only entry there's no need for a map yet.